File: src/main/java/bad/robot/radiate/ui/FadeIn.java
package bad.robot.radiate.ui;

import java.beans.PropertyChangeListener;
import java.util.Arrays;

import static java.lang.String.format;
import static java.util.Arrays.*;

@Deprecated
class FadeIn implements Fade {

    private float limit = 50f;
    private float count = 0f;

    @Override
    public void fireEvent(PropertyChangeListener... listeners) {
        if (++count <= limit) {
            stream(listeners).forEach(listener -> listener.propertyChange(new AlphaTransparencyChangeEvent(this, (count - 1) / limit, count / limit)));
        }
    }

    @Override
    public boolean done() {
        return count == limit;
    }

    @Override
    public String toString() {
        return format("%.2f", count / limit);
    }
}

File: src/main/scala/bad/robot/radiate/ui/FadeIn.scala
package bad.robot.radiate.ui

import java.beans.PropertyChangeListener

class FadeIn extends Fade {

  private val limit = 50f

  private var count = 0f

  def fireEvent(listeners: List[PropertyChangeListener]) {
    count += 1
    if (count <= limit) {
      listeners.foreach(_.propertyChange(new AlphaTransparencyChangeEvent(this, (count - 1) / limit, count / limit)))
    }
  }

  def done = count == limit

  override def toString = f"${count / limit}%.2f"
}